course-mw/course-actors/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java

```diff
@@ -97,55 +97,76 @@ private void createCourseBatch(Request actorMessage) throws Throwable {
     List<Map<String, Object>> correlatedObject = new ArrayList<>();
     String courseBatchId = ProjectUtil.getUniqueIdFromTimestamp(actorMessage.getEnv());
     Map<String, String> headers =
-        (Map<String, String>) actorMessage.getContext().get(JsonKey.HEADER);
+            (Map<String, String>) actorMessage.getContext().get(JsonKey.HEADER);
     String requestedBy = (String) actorMessage.getContext().get(JsonKey.REQUESTED_BY);
 
     if (Util.isNotNull(request.get(JsonKey.PARTICIPANTS))) {
       ProjectCommonException.throwClientErrorException(
-          ResponseCode.invalidRequestParameter,
-          ProjectUtil.formatMessage(
-              ResponseCode.invalidRequestParameter.getErrorMessage(), PARTICIPANTS));
+              ResponseCode.invalidRequestParameter,
+              ProjectUtil.formatMessage(
+                      ResponseCode.invalidRequestParameter.getErrorMessage(), PARTICIPANTS));
     }
     CourseBatch courseBatch = JsonUtil.convert(request, CourseBatch.class);
     courseBatch.setStatus(setCourseBatchStatus(actorMessage.getRequestContext(), (String) request.get(JsonKey.START_DATE)));
     String courseId = (String) request.get(JsonKey.COURSE_ID);
-    Map<String, Object> contentDetails = getContentDetails(actorMessage.getRequestContext(),courseId, headers);
+    Map<String, Object> contentDetails = getContentDetails(actorMessage.getRequestContext(), courseId, headers);
     courseBatch.setCreatedDate(ProjectUtil.getTimeStamp());
-    if(StringUtils.isBlank(courseBatch.getCreatedBy()))
-    	courseBatch.setCreatedBy(requestedBy);
+    String primaryCategory = (String) contentDetails.getOrDefault(JsonKey.PRIMARYCATEGORY, "");
+    if (StringUtils.isBlank(courseBatch.getCreatedBy()))
+      courseBatch.setCreatedBy(requestedBy);
     validateContentOrg(actorMessage.getRequestContext(), courseBatch.getCreatedFor());
     validateMentors(courseBatch, (String) actorMessage.getContext().getOrDefault(JsonKey.X_AUTH_TOKEN, ""), actorMessage.getRequestContext());
     courseBatch.setBatchId(courseBatchId);
+    if (primaryCategory.equalsIgnoreCase(JsonKey.PRIMARY_CATEGORY_BLENDED_PROGRAM) && courseBatch.getCurrentBatchSize() != null) {
+      try {
+        int maxUsersInBatch = Integer.parseInt(PropertiesCache.getInstance().getProperty(JsonKey.MAX_USERS_IN_BATCH));
+        if (maxUsersInBatch < courseBatch.getCurrentBatchSize()) {
+          ProjectCommonException.throwClientErrorException(
+                  ResponseCode.invalidCurrentBatchSize, ResponseCode.invalidCurrentBatchSize.getErrorMessage());
+        } else {
+          Map<String, Object> map = new HashMap<>();
+          map.put(JsonKey.CURRENT_BATCH_SIZE, String.valueOf(courseBatch.getCurrentBatchSize()));
```

Comment:
there is only one property - maxUsersInBatch which needs to be stored in batchAttributes upon creating a batch.
we don't need to read (from UI) nor store currentBatchSize into batchAttributes.